### ISSUE TITLE
Postgres serialization issue

### ISSUE THREAD

**Issue by Saddamus** [Chocobozzz/PeerTube#3607]
Hello. 
When traffic is reaching a decent number of concurrent viewers, we are noticing following errors:

Postgres: 
`  INSERT INTO "actor" ("id","type","preferredUsername","url","publicKey","privateKey","followersCount" │
│ ,"followingCount","inboxUrl","outboxUrl","sharedInboxUrl","followersUrl","followingUrl","createdAt","updatedAt") VALUES (DEFAULT,$1,$2,$3,$4,$5,$6, │
│ $7,$8,$9,$10,$11,$12,$13,$14) RETURNING "id","type","preferredUsername","url","publicKey","privateKey","followersCount","followingCount","inboxUrl" │
│ ,"outboxUrl","sharedInboxUrl","followersUrl","followingUrl","avatarId","serverId","createdAt","updatedAt"; 
could not serialize access due to read/write dependencies among transactions 
`

Peertube: 
` error: Error in controller. {
  "err": "SequelizeConnectionAcquireTimeoutError: Operation timeout\n    at ConnectionManager.getConnection (/app/node_modules/sequelize/lib/dialects/abstract/connection-manager.js:288:48)"`
Both database and peertube are running on kubernetes - powerful nodes. DId anybody faced this kind of issue - have any idea what can be done to get rid of these errors ?

**Comment by Chocobozzz** [comment]
Hi,

What is your postgresql version? And your instance URL?

**Comment by Saddamus** [comment]
PostgreSQL 11.10 on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit (bitnami helm chart)
https://watch.deranalyst.ch

**Comment by Chocobozzz** [comment]
What is a "decent number of concurrent viewers"? Do you have long running SQL queries when you see this kind of error?

**Comment by infinitron** [comment]
I came across the same error but in a different context. My setup has six live video streaming nodes and one node with peertube. The serialization error occurs whenever all the nodes start streaming at the same time, and the UI doesn't show live streams from 3-4 of them. I have a separate db setup exclusively for peertube, so there aren't any other queries being executed on the db. I added a random delay on each node before they started a live stream, and it solved my problem.
